pytest_fixtures/component/mcp.py

```diff
@@ -35,38 +35,93 @@ def _create_viewer_user(target_sat, org, location, permission_name):
+    assert (
+        target_sat.execute(f'firewall-cmd --permanent --add-port="{settings_obj.port}/tcp"').status
+        == 0
     )
-    sleep(20)
-    result = module_target_sat.execute(
-        f'podman inspect -f "{{{{.State.Status}}}}" {container_name}'
+    assert target_sat.execute('firewall-cmd --reload').status == 0
+
+    target_sat.ensure_podman_installed()
+    if is_downstream:
+        target_sat.podman_login(
+            settings_obj.registry_stage_username,
+            settings_obj.registry_stage_password,
+            settings_obj.registry_stage,
+        )
+        registry = settings_obj.registry_stage
+    else:
+        registry = settings_obj.registry
+
+    assert target_sat.execute(f'podman pull {registry}/{settings_obj.image_path}').status == 0
+
+    target_sat.execute(
+        f'podman run --name {container_name} -d --pull=never -it -p {settings_obj.port}:8080 \
+            {image_name} --foreman-url https://{target_sat.hostname} --host 0.0.0.0 --no-verify-ssl'
     )
-    log = module_target_sat.execute(f'podman logs {container_name}')
+    sleep(20)
+
+    result = target_sat.execute(f'podman inspect -f "{{{{.State.Status}}}}" {container_name}')
+    log = target_sat.execute(f'podman logs {container_name}')
     assert result.stdout.strip() == 'running', (
         f'failed to start container {container_name}: {log.stdout}'
         f'failed to start container {container_name}: {log.stderr}'
     )
+    return container_name
+
+
+def _cleanup_mcp_server(target_sat, container_name, registry, image_path):
+    """Helper function to clean up MCP server.
+
+    :param target_sat: The target satellite instance
+    :param container_name: Name of the container to clean up
+    :param registry: Registry path
+    :param image_path: Image path
+    """
+    target_sat.execute(f'podman kill {container_name}')
+    result = target_sat.execute(f'podman inspect -f "{{{{.State.Status}}}}" {container_name}')
+    assert result.stdout.strip() == 'exited', f'failed to clean up container {container_name}'
+    target_sat.execute(f'podman rm {container_name}')
+    target_sat.execute(f'podman rmi {registry}/{image_path}')
+
+
+@pytest.fixture(scope='module')
+def module_mcp_target_sat(module_target_sat):
+    """A module-level fixture to provide an MCP server configured on Satellite"""
+    container_name = _setup_mcp_server(module_target_sat, settings.foreman_mcp)
+    yield module_target_sat
+    _cleanup_mcp_server(
+        module_target_sat,
+        container_name,
+        settings.foreman_mcp.registry,
+        settings.foreman_mcp.image_path,
+    )
+
+
+@pytest.fixture(scope='module')
+def module_downstream_mcp_target_sat(module_target_sat):
+    """A module-level fixture to provide an downstream distribution of MCP server configured on Satellite"""
+    container_name = _setup_mcp_server(
+        module_target_sat, settings.foreman_mcp_downstream, is_downstream=True
```

Comment:
There is a method in our Capsule (and therefore Satellite) class to determine whether it is an upstream or downstream installation. Would it make sense to leverage it here?
https://github.com/SatelliteQE/robottelo/blob/517e422ae6b0f26dd62728d0bb867c8d925ce58a/robottelo/hosts.py#L1717-L1724